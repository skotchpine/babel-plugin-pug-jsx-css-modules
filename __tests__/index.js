const babel = require('babel-core')
const plugin = require('../')

const examples = [`
  const Styles = {h1: {fontSize: 32}}
  const X = ({name}) => <Text className="h1">name</Text>
`,`
  const Styles = {h1: {fontSize: 32}}
  const X = ({name}) => <Text className="h1" style={{fontSize: 42}}>name</Text>
`,`
  const Styles = {h1: {fontSize: 32}, h2: {fontSize: 24}, 'flex-1': {flex: 1}}
  const X = ({name}) => <Text className="h1 h2 flex-1" style={{fontSize: 42}}>name</Text>
`]

examples.forEach(example => {
  it('injects styles from classNames', () => {
    const {code} = babel.transform(example, {plugins: [plugin], presets: ['@babel/preset-env', '@babel/preset-react']})
    expect(code).toMatchSnapshot()
  })
})
