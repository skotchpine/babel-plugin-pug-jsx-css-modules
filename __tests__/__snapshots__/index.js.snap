// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`injects styles from classNames 1`] = `
"\\"use strict\\";

var Styles = {
  h1: {
    fontSize: 32
  }
};

var X = function X(_ref) {
  var name = _ref.name;
  return /*#__PURE__*/React.createElement(Text, {
    style: [Styles[\\"h1\\"]]
  }, \\"name\\");
};"
`;
